tests/foreman/ui/test_containerimagetag.py

```diff
@@ -27,12 +27,16 @@
 
 @pytest.fixture(scope="module")
 def module_org():
-    return entities.Organization().create()
+    org = entities.Organization().create()
+    yield org
+    org.delete()
```

Comment:
Deleting orgs can cause race conditions, this is not safe. Deleting anything just waits time given we do not keep the Satellite after testing.

Reply:
When the org is module scoped, what is the race condition this test module would hit during module teardown?

Reply:
I think there is a happy medium here, as the test will only fail when run multiple times for either product or repo, not organization, so we can remove the yield for organization and use the same org over consecutive runs. 

However, I think the point regarding race conditions should be discussed further by the team so that we have a clear standard going forward. In theory, each test should be agnostic to one another, meaning they are responsible for their own setup and teardown. For example, a test should not assume the org it is using is not being used by another test, rather, that test should specifically create or select an org that is specific that that test or module. I understand there is some nuance and tradeoffs that must be accounted for in the way we distribute tests in CI, so I think this is an important point that should be worked out by the team so we have a more clear standard going forward